infra/test/Test/Pos/Infra/Arbitrary/Ssc.hs

```diff
@@ -0,0 +1,282 @@
+import           Test.QuickCheck.Arbitrary.Generic (genericArbitrary,
+                     genericShrink)
+
+import           Pos.Core (EpochIndex, SlotId (..), VssCertificate (..),
+                     VssCertificatesMap, mkVssCertificate,
+                     mkVssCertificatesMapLossy)
+import           Pos.Core.Configuration (HasProtocolConstants,
+                     protocolConstants)
+import           Pos.Core.ProtocolConstants (ProtocolConstants (..),
+                     VssMaxTTL (..), VssMinTTL (..))
+import           Pos.Core.Ssc (Commitment (..), CommitmentsMap, Opening (..),
+                     SignedCommitment, SscPayload (..), SscProof (..),
+                     mkCommitmentsMap, randCommitmentAndOpening)
+import           Pos.Crypto (ProtocolMagic, SecretKey, deterministic,
+                     randomNumberInRange, toVssPublicKey, vssKeyGen)
+import           Pos.Infra.Communication.Types.Relay (DataMsg (..))
+import           Pos.Sinbin.Ssc.Base (isCommitmentIdExplicit,
+                     isOpeningIdExplicit, isSharesIdExplicit,
+                     mkSignedCommitment)
+import           Pos.Sinbin.Ssc.Message (MCCommitment (..), MCOpening (..),
+                     MCShares (..), MCVssCertificate (..), SscTag (..))
+import           Pos.Sinbin.Ssc.Toss.Types (TossModifier (..))
+import           Pos.Sinbin.Ssc.Types (SscGlobalState (..),
+                     SscSecretStorage (..))
+import           Pos.Sinbin.Ssc.VssCertData (VssCertData (..))
+
+import           Test.Pos.Core.Arbitrary (genVssCertificate)
+import           Test.Pos.Core.Arbitrary.Unsafe ()
+import           Test.Pos.Crypto.Arbitrary (genSignature)
+import           Test.Pos.Crypto.Dummy (dummyProtocolMagic)
+import           Test.Pos.Util.QuickCheck.Arbitrary (Nonrepeating (..),
+                     sublistN)
+
+----------------------------------------------------------------------------
+-- Types
+----------------------------------------------------------------------------
+
+newtype SscPayloadDependsOnSlot = SscPayloadDependsOnSlot
+    { genPayloadDependsOnSlot :: SlotId -> Gen SscPayload
+    } deriving Generic
+
+----------------------------------------------------------------------------
+-- Core
+----------------------------------------------------------------------------
+
+-- | Wrapper over 'SignedCommitment'. Creates an invalid SignedCommitment
+-- w.r.t. 'verifyCommitmentSignature'.
+newtype BadSignedCommitment = BadSignedComm
+    { getBadSignedC :: SignedCommitment
+    } deriving (Generic, Show, Eq)
+
+instance Arbitrary BadSignedCommitment where
+    arbitrary = BadSignedComm <$> do
+        pk <- arbitrary
+        sig <- arbitrary
+        comm <- Commitment <$> arbitrary <*> arbitrary
+        return (pk, comm, sig)
+    shrink = genericShrink
+
+-- | Pair of 'Commitment' and 'Opening'.
+data CommitmentOpening = CommitmentOpening
+    { coCommitment :: !Commitment
+    , coOpening    :: !Opening
+    } deriving (Generic, Show)
```

Comment:
Why is this duplicated from [here](https://github.com/input-output-hk/cardano-sl/blob/erikd/CDEC-433-2/ssc/test/Test/Pos/Ssc/Arbitrary.hs#L94)?

Reply:
Good catch! I think a l messed up when I was disentangling this mess.

The fix is that `Test.Pos.Infra.Arbitrary.Ssc` now imports `Test.Pos.Ssc.Arbitrary` and I initially imported it unqualified and open so GHC screamed about all the duplication until I fixed them all. I then made the import of `Test.Pos.Ssc.Arbitrary` explicit.